openbb_platform/providers/oecd/openbb_oecd/models/gdp_forecast.py

```diff
@@ -94,11 +96,20 @@ def extract_data(
         )
         data_df = data_df[data_df["country"] == query.country]
         data_df = data_df[["country", "date", "value"]]
-        return data_df.to_dict(orient="records")
+        data = data_df.to_dict(orient="records")
+        start_date = query.start_date.strftime("%Y-%m-%d")  # type: ignore
+        end_date = query.end_date.strftime("%Y-%m-%d")  # type: ignore
+        data = [
+            x
+            for x in data
+            if f"{x['date']}-01-01" >= start_date and f"{x['date']}-12-31" <= end_date
```

Comment:
This is funky.

![Screenshot 2024-02-29 at 12 43 07 PM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/28e5319b-4572-4872-8d9a-8b467adcec7b)

Reply:
You sent screenshot of STIR, but commented on gdp_forecast.  

This gdp was a quick hack because it was returning just a year and itll need to be updated anyways as stats. is being deprecated soon. Will try to figure why your screenshot isnt working

Reply:
Aha its because it hasnt been validated yet.

```
{'country': 'ireland', 'date': '1984-02', 'value': 12.69}
```

This returns YYYY-mm so it cant exactly filter.  This means I need to get creative.  Hang tight.

Reply:
We can simplify the date validator, no need to regex it.

```
if "Q" in str(in_date):
    return to_datetime(in_date).to_period("Q").to_timestamp("Q").date()
if len(str(in_date) == 7:
    return pd.to_datetime(in_date).to_period("M").to_timestamp("M").date()
```

Reply:
yeah thats smarter.